cpp/src/arrow/dataset/scanner.cc

```diff
@@ -843,6 +834,24 @@ Status ScannerBuilder::BatchSize(int64_t batch_size) {
   return Status::OK();
 }
 
+Status ScannerBuilder::BatchReadahead(int32_t batch_readahead) {
+  if (batch_readahead < 0) {
```

Comment:
What does it mean if `batch_readahead == 0`?

Reply:
Does this mean that you just won't read ahead?